### PR TITLE
Kazoo websockets

### PR DESCRIPTION
**Pull request by macpie**

### CODE REVIEW

File: applications/crossbar/src/modules/cb_websockets.erl [hunk 1]
```diff
@@ -0,0 +1,243 @@
...
+         ,validate/1, validate/2
+        ]).
+
+-include("../crossbar.hrl").
+
+-define(CB_LIST, <<"websockets/crossbar_listing">>).
+-define(
+    AVAILABLE
+    ,wh_json:from_list([
+        {<<"call">>, [<<"call.CHANNEL_CREATE.*">>, <<"call.CHANNEL_ANSWER.*">>, <<"call.CHANNEL_DESTROY.*">>]}
```
Comment (lazedo) [inline applications/crossbar/src/modules/cb_websockets.erl]: what about fax status ?

Comment (macpie) [inline applications/crossbar/src/modules/cb_websockets.erl]: yep missed that thank you

**Comment by macpie** [comment]
@lazedo @jamesaimonetti  any idea why?

blackhole_tracking.erl:178: Record construction #bh_context{auth_token::'_',auth_account_id::'_',bindings::'_',websocket_session_id::'_',websocket_pid::'_',req_id::'_',timestamp::'_',name::'_',metadata::'_',destination::'_',source::'_'} violates the declared type of field auth_token::'undefined' | binary() and auth_account_id::'undefined' | binary() and account_id::'undefined' | binary() and bindings::[<<_:8,_:_*8>>] and websocket_session_id::'undefined' | binary() and websocket_pid::'undefined' | pid() and req_id::<<_:8,_:_*8>> and timestamp::'undefined' | pos_integer() and name::'undefined' | binary() and destination::'undefined' | binary() and source::'undefined' | binary()

**Comment by fenollp** [comment]
@macpie `#bh_context{}`'s record definition needs to be allowed to take `'_'`.

File: applications/crossbar/src/modules/cb_websockets.erl [hunk 2]
```diff
@@ -0,0 +1,244 @@
...
+         ,validate/1, validate/2
+        ]).
+
+-include("../crossbar.hrl").
+
+-define(CB_LIST, <<"websockets/crossbar_listing">>).
+-define(
+    AVAILABLE
+    ,wh_json:from_list([
+        {<<"call">>, [<<"call.CHANNEL_CREATE.*">>, <<"call.CHANNEL_ANSWER.*">>, <<"call.CHANNEL_DESTROY.*">>, <<"call.RECORD_STOP.*">>]}
```
Comment (lazedo) [inline applications/crossbar/src/modules/cb_websockets.erl]: <<"call.RECORD_STOP.*">> is this actually supported ?
also, there was a recent addition to wh_hooks_listener for CHANNEL_BRIDGE, which gets supported here automagically.

Comment (macpie) [inline applications/crossbar/src/modules/cb_websockets.erl]: Adding it

**Comment by lazedo** [comment]
@macpie did you test the RECORD_STOP event  binding ? i believe it doesn't work since call events are binded to wh_hooks_listener, or i'm missing something here.

**Comment by macpie** [comment]
This was in the old doc so I just assumed it should work. Nevertheless it is not required right now, so I can just remove it.

**Pull request merged**